api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgeConfigSpec defines the desired state of EdgeConfig
+type EdgeConfigSpec struct {
+	// Definition of the custom playbook workload to execute
+	EdgePlaybook *EdgePlaybookSpec `json:"edgePlaybook,omitempty"`
```

Comment:
What is the value of `EdgePlaybookSpec`?

Reply:
I'm not sure I understand your question/doubt correctly. 
If you're wondering why I created an `EdgePlaybookSpec` struct instead of putting the various fields directly into `EdgeConfigSpec` struct, it's because `EdgeConfig` is a generic CRD that can contain various types of configurations. At the moment we have only `EdgePlaybook` but in the future we could add more. So in this perspective, I preferred to create an `EdgePlaybookSpec` "sub-configuration". 
Did I answer your question? If not, could you help me to better understand what is your doubt?

Reply:
I understand and how we would model "various types of configurations"? Would we add more fields next to `EdgePlaybook`?

Reply:
Yes, that's what I had in mind. 
The structure will be 
```
type EdgeConfigSpec struct {
	EdgePlaybook *EdgePlaybookSpec 
        AnotherConfig  *AnotherConfigSpec
        ... etc ...
}
```

But if you think this isn't the best way to model additional configurations or if you think there might be drawbacks I'd be happy to get your opinion. 
PS. a similar argument applies to the `EdgePlaybookStatus`.